plotly/figure_factory/_bullet.py

```diff
@@ -0,0 +1,257 @@
+        be qualitative indicators against which the marker and measure bars
+        are compared.
+        Default=['rgb(198, 198, 198)', 'rgb(248, 248, 248)']
+    :param (list) measure_colors: a list of two colors which is used to color
+        the thin quantitative bars in the bullet chart.
+        Default=['rgb(31, 119, 180)', 'rgb(176, 196, 221)']
+    :param (float) subplot_spacing: set the distance between each bar chart.
+        If not specified an automatic spacing is assigned based on the number
+        of bars to be plotted.
+    :param (str) title: title of the bullet chart.
+    :param (float) height: height of the chart.
+    :param (float) width width of the chart.
+    """
+    # validate df
+    if not pd:
+        raise exceptions.ImportError(
+            "'pandas' must be imported for this figure_factory."
+        )
+
+    if isinstance(df, list):
+        if not all(isinstance(item, dict) for item in df):
+            raise exceptions.PlotlyError(
+                'If your data is a list, all entries must be dictionaries.'
+            )
+        df = pd.DataFrame(df)
+
+    elif not isinstance(df, pd.DataFrame):
+        raise exceptions.PlotlyError(
+            'You must input a pandas DataFrame or a list of dictionaries.'
+        )
+
+    # check for valid keys
+    if any(key not in VALID_KEYS for key in df.columns):
+        raise exceptions.PlotlyError(
+            'Your headers/dict keys must be either {}'.format(
+                utils.list_of_options(VALID_KEYS, 'or')
+            )
+        )
+
+    # add necessary columns if missing
+    for key in VALID_KEYS:
+        if key not in df:
+            if key in ['title', 'subtitle']:
+                element = ''
+            else:
+                element = []
+            df[key] = [element for _ in range(len(df))]
+
+    # make sure ranges and measures are not NAN or NONE
+    for needed_key in ['ranges', 'measures']:
+        for idx, r in enumerate(df[needed_key]):
+            try:
+                r_is_nan = math.isnan(r)
+                if r_is_nan or r is None:
+                    df[needed_key][idx] = []
+            except TypeError:
+                pass
+
+    # validate custom colors
+    for colors_list in [range_colors, measure_colors]:
+        if colors_list:
+            if len(colors_list) != 2:
+                raise exceptions.PlotlyError(
+                    "Both 'range_colors' or 'measure_colors' must be a list "
+                    "of two valid colors."
+                )
+            colors.validate_colors(colors_list)
+            colors_list = colors.convert_colors_to_same_type(colors_list,
+                                                             'rgb')[0]
+
+    fig = _bullet(
+        df, as_rows, marker_size, marker_symbol, range_colors, measure_colors,
+        subplot_spacing
+    )
+
+    fig['layout'].update(
+        title=title,
+        height=height,
+        width=width,
```

Comment:
if this was `layout_options` then you could just pass in `fig['layout'].update(layout_options)`